CreateSurfaceWindowsImage.ps1

```diff
@@ -1022,100 +1107,92 @@ Function Get-LatestSurfaceEthernetDrivers
+        $global:KBGUID = $guids[0]
+
+        $scriptblock = {
+            $guid = $_.Guid
+            $itemtitle = $_.description
+            $guid
+        
+            $post = @{ size = 0; updateID = $guid; uidInfo = $guid } | ConvertTo-Json -Compress
+            $body = @{ updateIDs = "[$post]" }
+            Invoke-WebRequest -Uri 'https://www.catalog.update.microsoft.com/DownloadDialog.aspx' -Method Post -Body $body | Select-Object -ExpandProperty Content
+        }
 
-        If ($array.count -gt 0)
-        {
-            $global:KBGUID = $array | Where-Object {($_.description -like "*Surface - Net - 10.*")}
+        $downloaddialogs = $global:KBGUID | ForEach-Object -Process $scriptblock
+        $updatesFound = $false
 
-            If ($global:KBGUID.Count -gt 1)
+        ForEach ($downloaddialog in $downloaddialogs)
+        {
+            $title = Get-DownloadDialogText -Text $downloaddialog -Pattern 'enTitle ='
+            If (!($title))
             {
-                $largest = ($global:KBGUID | Measure-Object -Property description -Maximum)
-                $global:KBGUID = $global:KBGUID | Where-Object {$_.description -eq $largest.Maximum}
+                #do nothing
             }
-        }
-
-        ForEach ($Object in $global:KBGUID)
-        {
-            $kb = $Object.kbId
-            $curTxt = $Object.description
-    
-            ##Create Post Request to get the Download URL of the Update
-            $Post = @{ size = 0; updateID = $kb; uidInfo = $kb } | ConvertTo-Json -Compress
-            $PostBody = @{ updateIDs = "[$Post]" }
-    
-            ## Fetch and parse the download URL
-            $PostRes = (Invoke-WebRequest -Uri 'http://www.catalog.update.microsoft.com/DownloadDialog.aspx' -Method Post -Body $postBody).content
-            $DownloadLinks = ($PostRes | Select-String -AllMatches -Pattern "(http[s]?\://download\.windowsupdate\.com\/[^\'\""]*)" | Select-Object -Unique | ForEach-Object { [PSCustomObject] @{ Source = $_.matches.value } } ).source
-            If ($DownloadLinks)
+            Else
             {
-                If ($DownloadLinks.Count -gt 1)
+                $downloaddialog = $downloaddialog.Replace('www.download.windowsupdate', 'download.windowsupdate')
+                $DLWUDOTCOM = ($downloaddialog | Select-String -AllMatches -Pattern "(http[s]?\://download\.windowsupdate\.com\/[^\'\""]*)" | Select-Object -Unique | ForEach-Object { [PSCustomObject] @{ Source = $_.matches.value } } ).source
+                $DLDELDOTCOM = ($downloaddialog | Select-String -AllMatches -Pattern "(http[s]?\://dl\.delivery\.mp\.microsoft\.com\/[^\'\""]*)" | Select-Object -Unique | ForEach-Object { [PSCustomObject] @{ Source = $_.matches.value } } ).source
+
+                If ($DLWUDOTCOM)
                 {
-                    ForEach ($URL in $DownloadLinks)
+                "URL:     $link"

```

Comment:
What is this supposed to be printing? I don't think `$link` exists at this scope?

Reply:
@cluberti Definitely need your input on this one